superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndFilterSelect.tsx

```diff
@@ -217,23 +213,16 @@ const DndFilterSelect = (props: DndFilterSelectProps) => {
 
   const onClickClose = useCallback(
     (index: number) => {
-      if (confirmDeletion) {
-        const { confirmationText, confirmationTitle, triggerCondition } =
-          confirmDeletion;
-        if (triggerCondition(values[index], values)) {
-          confirm({
-            title: confirmationTitle,
-            content: confirmationText,
-            onOk() {
-              removeValue(index);
-            },
-          });
+      if (canDelete) {
+        const result = canDelete(values[index], values);
```

Comment:
Nit: If we did `canDelete?.(...)`, we could get rid of 1 level of indentation (`canDelete` check)

Reply:
👍🏼